test/metabase/query_processor_test/date_time_zone_functions_test.clj

```diff
@@ -292,6 +292,48 @@
           (testing (format "%s %s function works as expected on %s column for driver %s" op unit col-type driver/*driver*)
             (is (= (set expected) (set (test-datetime-math query))))))))))
 
+(defn- close? [t1 t2 period]
+  (and (t/before? (t/instant t1) (t/plus (t/instant t2) period))
+       (t/after? (t/instant t1) (t/minus (t/instant t2) period))))
+
+(deftest now-test
+  (mt/test-drivers (mt/normal-drivers-with-feature :date-arithmetics)
```

Comment:
Something lurking in my head. The mongo driver supports date-arithmetics conditioned on the version:

```clojure
(defmethod driver/database-supports? [:mongo :date-arithmetics] [_ _ db]
  (let [version (db-major-version db)]
    (and (some? version) (>= version 5))))
```

But it of course does not use the `sql.query-processor` and therefore does not offer an implementation for `current-datetime-honeysql-form`.  The "normal" drivers would include this

```clojure
(def ^:private abnormal-drivers
  "Drivers that are so weird that we can't run the normal driver tests against them."
  #{:druid :googleanalytics})

(defn normal-drivers
  "Drivers that are reasonably normal in the sense that they can participate in the shared driver tests."
  []
  (set/difference (tx.env/test-drivers) abnormal-drivers))
```

Seems like we need to specify a bit more which drivers support this in a more targeted manner. Or (perhaps follow up work) to move this method into the query processor rather than the sql.query-processor?

Reply:
> Seems like we need to specify a bit more which drivers support this in a more targeted manner.

Indeed. I think this example justifies a `:now` feature flag, despite the discussion. We already have feature flags per function, e.g. `convert-timezone`. I'll change it over. 

Moving `sql.qp/current-datetime-honeysql-form` from sql.query-processor and have it be implemented by non-sql drivers makes total sense. mongo currently cheats with :relative-datetime and uses the current time from clojure, not the time in the database transaction.
https://github.com/metabase/metabase/blob/bfa7c42e4312ba7fc02ab731cdbf7cccca9b0b2a/modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj#L356

I'll change this PR so :now has it's own feature flag, so at least 3rd party drivers can upgrade without implementing it if they wish.

I'll also update mongo so :now is implemented for all versions. I don't know how that passed the tests.